### DISCUSSION TITLE
Serial formatting

### DISCUSSION THREAD

**Discussion by jocundjackson** [collin80/SavvyCAN#651] [Q&A]
Hello, what format is expected from the serial output? 

I appear to have something being seen from SavvyCAN via the device console(LAWICEL / SLCAN Serial), but I wonder if it is in the wrong format as no data is showing in the main SavvyCAN window. The serial device goes silent when unplugged from USB. When the CANbus is silent, so is the device console output. The output from my Arduino is appearing in the Arduino IDE serial console has pretty-printed can messages, in the format `ID, D0 D1 D2 ...` so I know I am receiving data properly, but I don't know what format is expected for SavvyCAN to identify those messages as CAN traffic.

Example SavvyCAN device console data stream:
<img width="352" alt="Screenshot 2023-07-09 at 18 39 16" src="https://github.com/collin80/SavvyCAN/assets/39113242/9f76925f-0eec-4d73-a341-18f180520952">

Thank you for your help!

**Comment by jocundjackson** [comment]
For future reference, the format which is expected to see the following hex values:
74 - "t" in ASCII encoding (start of message)
d - 13 in decimal (end of message/carriage return in ASCII, 0x13)

The data should be passed in the following format:
3 bytes of frame ID
1 byte of data length
n bytes of data, determined by data length.

**Reply by EijiTomonari** [reply]
Would you please share a full message example?
